### ISSUE TITLE
only-arrow-functions: false negative when a function is defined in an ES6 module

### ISSUE THREAD

**Issue by slawomir** [Microsoft/tslint-microsoft-contrib#290]
_// module.ts_
```
// ES6 module
export function foo(): void {
    return undefined;
}
```
_// main.ts_
```
// ES import
import * as Module from 'Module';
```

**Expected**
It should be allowed, since it's a way for declaring modules in ES6

**Actual**
Error: `non-arrow functions are forbidden`

Reference: https://palantir.github.io/tslint/rules/no-namespace/

**Comment by HamletDRC** [comment]
@slawomir only-arrow-functions is a tslint rule. Can you create this bug report over on the tslint website? 

https://github.com/palantir/tslint/issues/new

**Comment by slawomir** [comment]
@HamletDRC I confused that with 'no-function-expression'. Has tslint's 'only-arrow-functions' the same purpose as 'no-function-expression'?

**Comment by HamletDRC** [comment]
yes, we wrote our first and then they implemented something similar. However, our rule is more intelligent around the 'this' reference. Sometimes you need a function in order to access 'this' of that function. Our rule will not fail in this case but theirs will. 

Do you see the false positive in no-function-expression or only-arrow-functions?

**Comment by slawomir** [comment]
Thanks for clarification @HamletDRC. I agree it makes sense to keep both version, since these are not fully equal. I didn't find any problems with no-function-expression. Bug for only-arrow-functions is now in here: https://github.com/palantir/tslint/issues/1655
